#646 Create training team feedback system

kjporter opened the pull request:
This PR will:
* Creates the TA-requested training team feedback feature
* Mirrors much of the controller feedback feature, but tailored for the training team
* Close #646

iccowan commented on app/Http/Controllers/AdminDash.php:
```diff
@@ -1045,6 +1046,127 @@ public function emailFeedback(Request $request, $id) {
 
+    public function showTrainerFeedback() {
+        $users = User::with('roles')->where('status', '1')->get();
+        $ins = $users->filter(function ($user) {
+            return $user->hasRole('ins');
+        });
+
+        $mtr = $users->filter(function ($user) {
+            return $user->hasRole('mtr');
+        });
+        $feedbackOptions = $ins->merge($mtr);
```
This would be cleaner and more efficient with:

```php
$feedbackOptions = $users->filter(function ($user) {
  return $user->hasRole('ins') || $user->hasRole('mtr');
});
```

kjporter replied:
Done

iccowan commented on app/Http/Controllers/FrontController.php:
```diff
@@ -434,6 +434,22 @@ public function saveNewFeedback(Request $request) {
 
+    public function newTrainerFeedback() {
+        $users = User::with('roles')->where('status', '1')->get();
+        $ins = $users->filter(function ($user) {
+            return $user->hasRole('ins');
+        });
+
+        $mtr = $users->filter(function ($user) {
+            return $user->hasRole('mtr');
+        });
+        $feedbackOptions = $ins->merge($mtr);
```
See above. Since we are doing the same thing twice, maybe we should retrieve the feedback options from the `TrainerFeedback` model as a static function so we don't have duplicated code

kjporter replied:
Done

iccowan commented on app/Http/Controllers/TrainingDash.php:
```diff
@@ -734,6 +735,67 @@ public function generateGraphs(Request $request) {
     }
 
+    public function newTrainerFeedback() {
+        $users = User::with('roles')->where('status', '1')->get();
+        $ins = $users->filter(function ($user) {
+            return $user->hasRole('ins');
+        });
+
+        $mtr = $users->filter(function ($user) {
+            return $user->hasRole('mtr');
+        });
+        $feedbackOptions = $ins->merge($mtr);
```
3rd time this exists, so let's definitely do this elsewhere and reuse it

kjporter replied:
Done

iccowan commented on resources/views/inc/notifications.blade.php:
We should really rework this file at some point and make it more readable, lol

kjporter replied:
Agreed - I was thinking the same thing

iccowan commented on resources/views/inc/trainer_feedback.blade.php:
Love this 💪

iccowan reviewed (changes requested):
Really nice work, just some places where the code could be a little less spaghetti-y

iccowan reviewed (commented):
Code looks good - I don't have time to QA now, but will get to it in the next few days

iccowan commented:
1. Saving feedback redirects to home page on main site, not controller dashboard
2. Updating saved feedback returns an error
3. Viewing anonymous feedback in the manage page is a little weird
4. Public link (https://qa-2.ztlartcc.org/trainer_feedback/new) does not work - `App\Http\Controllers\TrainerFeedback not found`

kjporter commented:
> 1. Saving feedback redirects to home page on main site, not controller dashboard
> 2. Updating saved feedback returns an error
> 3. Viewing anonymous feedback in the manage page is a little weird
> 4. Public link (https://qa-2.ztlartcc.org/trainer_feedback/new) does not work - `App\Http\Controllers\TrainerFeedback not found`

@iccowan thanks - those have all been corrected.

iccowan commented:
When saving from the public site, it logs in the user and does not save the ticket. There may be some kind of authentication mixup happening on the backend

kjporter commented:
> When saving from the public site, it logs in the user and does not save the ticket. There may be some kind of authentication mixup happening on the backend

Fixed - it was a routing issue. Moved the save routine route so it will work properly.
